reader.go

```diff
@@ -151,65 +179,67 @@ func (rd *Reader) readHeader() (*Header, error) {
 		rd.pad = 0
 	}
 
-	// Handle long filenames
-	if header.Name[0] == '/' && rd.longFilenames != nil {
-		if offset, err := strconv.Atoi(header.Name[1:]); err == nil {
-			data := rd.longFilenames[offset:]
-			if idx := bytes.IndexByte(data, '\n'); idx != -1 {
-				data = data[:idx]
-			}
-			if idx := bytes.IndexByte(data, '/'); idx != -1 {
-				data = data[:idx]
-			}
-			header.Name = string(data)
+	// The ar format only supports file names of up to 16 bytes to be stored in the file name header
+	// field. Various archivers have adopted different conventions for storing files with longer
+	// names:
+	// - GNU ar replaces the file name in the header with the character "/" followed by an integer. The
```

Comment:
Given this list format, I'd expect more than one entry?

Reply:
Oh, I see the other entry is further down. That's mildly confusing. At the very least I think this "GNU ar" entry should go with the `else if header.Name[0] == '/'`?